### DISCUSSION TITLE
BROADCASTING 模式下怎么看消费速度和延迟

### DISCUSSION THREAD

**Discussion by wz2cool** [apache/rocketmq#8094] [Q&A]
请问如何看 BROADCASTING 每个消费者的消费速度 和延迟情况

**Comment by cserwen** [comment]
广播消费这部分没有目前没有监控，所以暂时看不了。不过可以通过一些特殊的方法实现一下这个功能，比如
- 可以通过 client 拿到所有分区的最大位点，然后读取本地保存的已提交的位点来计算堆积；
- 定时读取本地已提交的位点来计算消费速度

**Reply by cserwen** [reply]
Java 消费者里也有一个 类用于记录了些状态，ConsumerStatsManager，但是没有堆积的信息，
```
ConsumerStatsManager stats = consumer.getDefaultMQPushConsumerImpl().getConsumerStatsManager()
```

**Reply by wz2cool** [reply]
感谢 我试试看哈
